google/cloud/storage/oauth2/service_account_credentials_test.cc

```diff
@@ -125,15 +132,32 @@ TEST_F(ServiceAccountCredentialsTest,
             return t;
           }));
 
-  auto info = ParseServiceAccountCredentials(kJsonKeyfileContents, "test");
-  ASSERT_TRUE(info.ok()) << "status=" << info.status();
   ServiceAccountCredentials<MockHttpRequestBuilder, FakeClock> credentials(
-      *info);
+      info);
   // Calls Refresh to obtain the access token for our authorization header.
   EXPECT_EQ("Authorization: Type access-token-value",
             credentials.AuthorizationHeader().value());
 }
 
+/// @test Verify that we can create service account credentials from a keyfile.
+TEST_F(ServiceAccountCredentialsTest,
+       RefreshingSendsCorrectRequestBodyAndParsesResponse) {
+  auto info = ParseServiceAccountCredentials(kJsonKeyfileContents, "test");
+  ASSERT_TRUE(info.ok()) << "status=" << info.status();
+  CheckInfoYieldsExpectedAssertion(*info, kExpectedAssertionParam);
+}
+
+/// @test Verify that we can create service account credentials from a keyfile.
+TEST_F(ServiceAccountCredentialsTest,
+       RefreshingSendsCorrectRequestBodyAndParsesResponseForNonDefaultVals) {
+  auto info = ParseServiceAccountCredentials(kJsonKeyfileContents, "test");
+  ASSERT_TRUE(info.ok()) << "status=" << info.status();
+  info->scopes = std::set<std::string>{kAltScopeForTest};
+  info->subject = google::cloud::optional<std::string>(kSubjectForGrant);
```

Comment:
`optional` *shouldn't* make things much more difficult or verbose to use. I think you can just directly assign here (and elsewhere):

```cc
info->subject = kSubjectForGrant;
```

Reply:
Done -- had to add a string constructor around it, as it didn't like trying to convert between a const char array and a string (which is why I tacked on the verbose optional wrapper before).